nntrainer/src/pooling2d_layer.cpp

```diff
@@ -156,8 +204,13 @@ Tensor Pooling2DLayer::pooling2d(Tensor in, int &status) {
           for (unsigned int pi = 0; pi < p_height; ++pi) {
             for (unsigned int pj = 0; pj < p_width; ++pj) {
               float val = in.getValue(0, i, j + pi, k + pj);
-              if (max < val)
+              if (max < val) {
+                max_idx[base_idx +
+                        i * output_dim.height() * output_dim.width() +
+                        I * output_dim.width() + J] =
+                  i * height * width + (j + pi) * width + (k + pj);
```

Comment:
Could you double check this?
```suggestion
              if (max < val) {
                max_idx[base_idx +
                        i * output_dim.height() * output_dim.width() +
                        I * output_dim.width() + J] =
                 batch * input_dim.getFeatureLen() + i * height * width + (j + pi) * width + (k + pj);
```

Reply:
Yes. this must be batch*input_dim.getFeatureLen(). Thanks.